src/app/shared/directive/clipboard.directive.ts

```diff
@@ -13,14 +13,37 @@ export class ClipboardDirective {
   @HostListener('click', ['$event'])
   public onClick(event: MouseEvent): void {
     event.preventDefault();
-    if (!this.payload || !navigator.clipboard) {
-      return;
+
+    if (!this.payload) return;
+
+    if (navigator.clipboard) {
+      this.copyUsingClipboardAPI();
+    } else {
+      this.copyUsingFallbackMethod();
     }
-    navigator.clipboard.writeText(this.payload.toString()).then(() => {
-      this.copied.emit(this.payload.toString());
-    }, (err) => {
-      this.copied.emit('Error could not copy text: ' + JSON.stringify(err));
-    });
   }
 
+  private copyUsingFallbackMethod(): void {
+    const input = document.createElement('textarea');
+    input.innerHTML = this.payload;
+    document.body.appendChild(input);
+    input.select();
+
+    try {
+      const result = document.execCommand('copy');
+      if (result) {
+        this.copied.emit(this.payload.toString());
+      } else {
+        this.copied.emit('Error could not copy text.');
+      }
+    } finally {
+      document.body.removeChild(input);
+    }
+  }
+
+  private copyUsingClipboardAPI(): void {
+    navigator.clipboard.writeText(this.payload.toString())
+      .then(() => this.copied.emit(this.payload.toString()))
+      .catch((error) => this.copied.emit('Error could not copy text: ' + JSON.stringify(error)));
```

Comment:
I'm open to suggestions for sanitization. I haven't looked at all places in the menus where copy is being used, for me the issue scope was not having copied invoices I created.

In the fallback, since there's gonna be a temp textarea element's content in the DOM it's important (as suggested above) to use `input.innerText` or `input.value`. I've opted for `innerText` and pushed the 1-line change.

Again, maybe a more simple alternative would be to not display the "copy" button when `navigator.clipboard` is not available.